samples/nrf9160/location/src/main.c

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <nrf_modem_at.h>
+#include <modem/lte_lc.h>
+#include <modem/location.h>
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <date_time.h>
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static struct location_config config;
+static struct location_method_config methods[3];
+
+K_SEM_DEFINE(location_event, 0, 1);
+
+static void antenna_configure(void)
```

Comment:
That's a fair question. We've probably discussed this earlier, but I don't remember if there was a good reason why this is handled by the sample and not by the library. If the application wants to use both Location library and GNSS API directly, then it's probably more logical that the configuration is handled by the application. This is the case with modem_shell, but that's a special case.

We could move this to the library, but it would have some implications for modem_shell. And I still hope that this LNA configuration is temporary and will eventually be moved to the board configuration.

Reply:
There at least one reason why moving it to `method_gnss.c` would be difficult. LNA needs to be configured before modem functional mode is set to 1, otherwise modem RF will not take the configuration into use. Location library init is most likely called too late, so it's better to keep this in the sample and eventually try to move it to the board configuration.